genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.scripts;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.genie.common.dto.JobRequest;
+import com.netflix.genie.web.exceptions.checked.ScriptExecutionException;
+import com.netflix.genie.web.exceptions.checked.ScriptNotConfiguredException;
+import com.netflix.genie.web.properties.ExecutionModeFilterScriptProperties;
+import io.micrometer.core.instrument.MeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Implementation of {@link ManagedScript} that delegates selection of a job's execution mode to a script,
+ * thus allowing fine-grained blacklisting/whitelisting when it comes to execution mode.
+ * See also: {@link com.netflix.genie.web.util.JobExecutionModeSelector}.
+ * <p>
+ * The contract between the script and the Java code is that the script will be supplied global variables
+ * {@code jobRequest} which will be JSON strings representing job request that kicked off this evaluation.
+ * The code expects the script to either return the true (to force agent execution), false (to force legacy/embedded
+ * execution) or null (for no preference).
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+public class ExecutionModeFilterScript extends ManagedScript {
```

Comment:
Shouldn't this be implementing some other interface reflecting that it is some sort of `ExecutionModeSelector`? Not sure if we have that for the Headers or Property based paths?

Reply:
This is just a script object. 
The execution mode is a different thing and uses this.